posthog/api/survey.py

```diff
@@ -251,7 +252,8 @@ def _create_or_update_targeting_flag(self, existing_flag=None, filters=None, nam
                 existing_flag_serializer.is_valid(raise_exception=True)
                 return existing_flag_serializer.save()
             elif name and filters:
-                feature_flag_key = slugify(f"{SURVEY_TARGETING_FLAG_PREFIX}{name}")
+                random_id = generate("1234567890abcdef", 10)
+                feature_flag_key = slugify(f"{SURVEY_TARGETING_FLAG_PREFIX}{random_id}{name}")
```

Comment:
there's a max char limit on the flag key, can we prune it + add a test for too long survey names? (sorry this should've existed already but 🤷 )

Reply:
👍🏾 I think I can just remove the survey name from the linked flag key. The description will have the survey name if necessary for lookup.

Reply:
I kinda like it for debugging, but also reasonable, specially since it can be wrong. Should've just used the survey ID in the first place.